new: Added lab on Helm (#967)
aws-samples/eks-workshop-v2: website/docs/introduction/helm/index.md

## Helm CLI

The `helm` CLI tool is typically used in conjunction with a Kubernetes cluster to manage the deployment and lifecycle of applications. It provides a consistent and repeatable way to package, install, and manage applications on Kubernetes, making it easier to automate and standardize application deployments across different environments.

The CLI is already installed in our IDE:

```bash
$ helm version
```

## Helm repositories

A Helm repository is a centralized location where Helm charts are stored and managed, and allow users to easily discover, share, and install charts. They facilitate easy access to a wide range of pre-packaged applications and services for deployment on Kubernetes clusters.

Lets add the `bitnami` repository to our Helm CLI:

```bash
$ helm repo add bitnami https://charts.bitnami.com/bitnami
$ helm repo update
```

Now we can search the repository for charts, for example the `postgresql` chart:

```bash
$ helm search repo postgresql
NAME                    CHART VERSION   APP VERSION     DESCRIPTION
bitnami/postgresql      X.X.X           X.X.X           PostgreSQL (Postgres) is an open source object-...
[...]
```

## Installing a Helm chart

> Lets install an NGINX server in our EKS cluster using the Helm chart we found above. When you install a chart using the Helm package manager, it creates a new **release** for that chart. Each release is tracked by Helm and can be upgraded, rolled back, or uninstalled independently from other releases.

Reviewer: ```suggestion
Let's install an NGINX server in our EKS cluster using the Helm chart we found above. When you install a chart using the Helm package manager, it creates a new **release** for that chart. Each release is tracked by Helm and can be upgraded, rolled back, or uninstalled independently from other releases.
```

```bash
$ echo $NGINX_CHART_VERSION
$ helm install nginx bitnami/nginx \
  --version $NGINX_CHART_VERSION \
  --namespace nginx --create-namespace --wait
```

We can break this command down as follows:

- Use the `install` sub-command to instruct Helm to install a chart
- Name the release `nginx`
- Use the chart `bitnami/nginx` with the version $NGINX_CHART_VERSION
- Install the chart in the `nginx` namespace and create that namespace first
- Wait for pods in the release to get to a ready state

Once the chart has installed we can list the releases in our EKS cluster:

```bash
$ helm list -A
NAME 	 NAMESPACE  REVISION  UPDATED                                  STATUS    CHART         APP VERSION
nginx	 nginx      1         2024-06-11 03:58:39.862100855 +0000 UTC  deployed  nginx-X.X.X   X.X.X
```

We can also see NGINX running in the namespace we specified:

```bash
$ kubectl get pod -n nginx
NAME                     READY   STATUS    RESTARTS   AGE
nginx-55fbd7f494-zplwx   1/1     Running   0          119s
```

## Configuring chart options

> In the example above we installed the NGINX chart in its default configuration. However its common to need to provide **values** to charts during installation to change different aspects of behavior of the installed component.

Reviewer: ```suggestion
In the example above we installed the NGINX chart in its default configuration. Sometimes you'll need to provide configuration **values** to charts during installation to modify the way the component behaves.
```

> You can provide values to charts during installation using various methods, with two common ways being:

Reviewer: ```suggestion
There are two common ways to provide values to charts during installation:
```
